### ISSUE TITLE
A11y_Azure Tools for VS Code_Azure Function_Select a runtime stack_Screenreader: Screenreader is announcing wrong names for the symbols.

### ISSUE THREAD

**Issue by kjawale99** [microsoft/vscode-azurefunctions#3124]
**GitHubTags:**

#A11yMAS; #A11yTCS; #A11ySev3; #DesktopApp; #Azure Tools for VS Code; #BM_Azure Tools for VS Code_Win32_Apr22; #Win32; #FTP; #NVDA; #JAWS; #Screenreader; #WCAG1.3.1

**Environment Details:**
Application Name: Visual Studio Code Services
Windows Version: Windows 11

**Repro Steps:**

1. Open Visual studio code
2. Tab till extension button and search for "Azure Functions"
3. Tab till install and and press enter now tab till the "Azure" button and press enter.
4. Right click on any of the functions
5. Tab navigate to "Create function app in azure" and hit enter.
6. Enter a global unique name for new function app and hit enter.
7. Select a runtime screen will appear.
8. Tab navigate to "Change Azure function version".
9. Turn on the JAWS.
10. Check if the JAWS reads the symbol correctly or not.

**Actual:**
Screenreader is announcing wrong names for "setting" symbol as "gear", "Add' symbol as "plus", "skip for now" symbol as "clock"

Note: Similar issue observed in NVDA.

**Expected Result:**
Screenreader should announce correct names for the symbols.

**Attachment**
![A11y_Azure Tools for VS Code_Azure Function_Select a runtime stack_Screenreader](https://user-images.githubusercontent.com/98745784/164449818-9886eb48-4384-4072-b988-6bf39b13bda5.gif)
:

**Comment by nturinski** [comment]
Could you please clarify what the issue here is?  Should the symbols not be read or is it actually be read incorrectly?  I can't discern exactly what your gif is trying to demonstrate.

Could you please give me a clear example of what the correct reading would be and what the incorrect reading currently is?

**Comment by nturinski** [comment]
After discussing this further, I think that the issue here is that the icons are being read as plain text.  From the listener's perspective, there is no difference between 🕑 and "clock"

**Comment by Ashutoshsahoo164** [comment]
Hi @nturinski , If those symbols are just for decorative purpose then screen reader should not announce anything related to it. Or if the symbol is conveying any information , then it should announce proper information . The Announcement "Gear" and "Clock" does not describe the symbols properly.

**Comment by MicroFish91** [comment]
VS Code team has recently added a feature request that extensions should be able to specify `accessibilityInformation` for `QuickPickItems`.  The implementation of this sort of feature should give us the flexibility we need to resolve issues like this one.  Here is a link to the issue where this was first discussed:  https://github.com/microsoft/vscode/issues/133169

**Comment by Ashutoshsahoo164** [comment]
Hi @MicroFish91 When can we expect this implementation will be done on the extension?

**Comment by MicroFish91** [comment]
Hi @Ashutoshsahoo164 I am not sure yet.  I can verify that they have added this feature request to their current backlog but they have not mentioned when they will prioritize this one.  I can say that they are a very diligent team and release regularly so we on the extensions team will be on the lookout to implement once this feature drops.  I know that's not a solid date, but I hope that helps some, thank you :)

**Comment by kjawale99** [comment]
@MicroFish91 The link for the issue which you gave is not similar to this issue.
Sev3 bugs do not have any defined SLA. Product team can act as per their Convenience.

**Comment by MicroFish91** [comment]
This one kind of fell off my radar.  Not sure the best way to address this.  There are some icons we can migrate to within this [list](https://code.visualstudio.com/api/references/icons-in-labels#icon-listing) that might be more descriptive, but there are other icons that don't really have good alternatives.

I dangled an upstream feature request [here](https://github.com/microsoft/vscode/issues/158541) to see if perhaps we could eventually put in custom screen reader descriptions.  If that's not feasible and there are not good icon alternatives, we will need to revisit and see if some of these icons should be removed entirely 🤷  

...TBD based on upstream feedback

**Comment by MicroFish91** [comment]
Based on upstream feedback listed in my comment directly above, we are back to tracking against this upstream issue:  https://github.com/microsoft/vscode/issues/133169

**Comment by MicroFish91** [comment]
Bumping this again, I believe this is resolved and can be closed, see upstream dialogue:

<img width="966" height="322" alt="Image" src="https://github.com/user-attachments/assets/7a936511-c16d-4984-af83-8f686854db41" />

**Comment by jinglouMSFT** [comment]
@kjawale99  Can you please resolve and close the bug?  Thanks.

**Comment by MicroFish91** [comment]
Closing as resolved, please see above.
